src/components/Embedded.js

```diff
@@ -29,11 +26,17 @@ const SUPPORTED_PANES = {
   result: true,
 };
 
+const savedState = state.load();
+
 // TODO: we should support readonly mode
 function Embedded() {
-  const [html, setHtml] = useState(savedState.markup || initialValues.html);
-  const [js, setJs] = useState(savedState.query || initialValues.js);
-  const { setParsed, htmlRoot } = useAppContext();
+  const [query, setQuery, markup, setMarkup] = usePlayground({
+    initialMarkup: initialValues.html || savedState.markup,
+    initialQuery: initialValues.js || savedState.js,
```

Comment:
should this not be `savedState.query`?

Maybe also rename the `html` and `js` properties on `initialValues`?

Reply:
makes sense